### ISSUE TITLE
False positive EmptyArrayAccess when array_flip is called on mixed type

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#2348]
I have warnings about missing return types disabled, and wouldn't expect this warning for unknown types.

Some other array builtins may be affected (filter or merge, not sure)

https://psalm.dev/r/f068bcb4af

`array_flip(include('path.php'))` is also affected

```php
<?php

class Test {
    public static function get_values() {
        return [
            1 => 'a',
            2 => 'b',
        ];
    }
    public static function find_value(string $key) {
        $map = array_flip(static::get_values());
        // Unexpected EmptyArrayAccess: Cannot access value on empty array variable $map
        return $map[$key];
    }
}
```

**Comment by TysonAndre** [comment]
The issue I mentioned for array_intersect_key is https://psalm.dev/r/41a2f58f02

```php
<?php
class Other {
    /** @psalm-suppress MissingReturnType */
    public static function unknown() {
        return ['x' => 'val'];
    }
}
class Intersect {
    /** @var array<string,string> */
    protected $s;
    public function sdn() {
        $r = array_intersect_key(Other::unknown(), array_filter($this->s));
        //  RedundantConditionGivenDocblockType - Found a redundant condition when evaluating docblock-defined type $r and trying to reconcile type 'array<empty, empty>' to empty
        return !empty($r) ? implode(',', $r) : '()';
    }
}
```
